lnwallet/channel.go

```diff
@@ -3195,18 +3195,30 @@ func (lc *LightningChannel) validateCommitmentSanity(theirLogCounter,
 		return ErrBelowChanReserve
 	}
 
-	// Calculate the commitment fee, and subtract it from the
-	// initiator's balance.
+	// Calculate the commitment fee, and subtract it from the initiator's
+	// balance.
 	commitFee := btcutil.Amount((int64(feePerKw) * commitWeight) / 1000)
+
+	// Before we apply the fee, we'll check to see if it's even payable
+	// using the initiator's balance. If the fee is greater than their
+	// current balance, then this commitment would consume their entire
+	// output, meaning the reservation would be violated.
+	switch {
+	case lc.channelState.IsInitiator && commitFee > ourBalance.ToSatoshis():
```

Comment:
Again, defining `commitFeeMSat` above would save some conversions here and below.

Reply:
Fixed (but left the conversions just to ensure we compare like kind to each other as that's what's manifested on the commitment transaction).